config/env_test.go

```diff
@@ -154,3 +157,210 @@ func TestOsdfEnvToPelican(t *testing.T) {
+	})
+
+	t.Run("job-ad-with-type-mismatch-string-to-int", func(t *testing.T) {
+		ResetConfig()
+
+		tmpDir := t.TempDir()
+		jobAdFile := filepath.Join(tmpDir, "test.job.ad")
+
+		// Try to set a string value where an int is expected
+		jobAdContent := `PelicanCfg_Client_MinimumDownloadSpeed = "not-a-number"
+`
+		err := os.WriteFile(jobAdFile, []byte(jobAdContent), 0644)
+		require.NoError(t, err)
+
+		os.Setenv("_CONDOR_JOB_AD", jobAdFile)
+		t.Cleanup(func() {
+			os.Unsetenv("_CONDOR_JOB_AD")
+		})
+
+		bindClassAdConfig()
+		_, err = param.Refresh()
+		// The refresh should fail because the value can't be parsed as an int
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "cannot parse")
+	})
+
+	t.Run("job-ad-with-int-where-bool-expected", func(t *testing.T) {
+		ResetConfig()
+
+		tmpDir := t.TempDir()
+		jobAdFile := filepath.Join(tmpDir, "test.job.ad")
+
+		// Try to set an integer where a bool is expected
+		jobAdContent := `PelicanCfg_Client_DisableHttpProxy = 1
+`
+		err := os.WriteFile(jobAdFile, []byte(jobAdContent), 0644)
+		require.NoError(t, err)
+
+		os.Setenv("_CONDOR_JOB_AD", jobAdFile)
+		t.Cleanup(func() {
+			os.Unsetenv("_CONDOR_JOB_AD")
+		})
+
+		bindClassAdConfig()
+		_, err = param.Refresh()
+		require.NoError(t, err)
+
+		// Viper should handle the conversion from int to bool (1 -> true, 0 -> false)
+		assert.Equal(t, true, param.Client_DisableHttpProxy.GetBool())
+	})
+
+	t.Run("job-ad-with-real-where-bool-expected", func(t *testing.T) {
+		ResetConfig()
+
+		tmpDir := t.TempDir()
+		jobAdFile := filepath.Join(tmpDir, "test.job.ad")
+
+		// Try to set a real number where a bool is expected
+		jobAdContent := `PelicanCfg_Client_DisableHttpProxy = 0.0
+`
+		err := os.WriteFile(jobAdFile, []byte(jobAdContent), 0644)
+		require.NoError(t, err)
+
+		os.Setenv("_CONDOR_JOB_AD", jobAdFile)
+		t.Cleanup(func() {
+			os.Unsetenv("_CONDOR_JOB_AD")
+		})
+
+		bindClassAdConfig()
+		_, err = param.Refresh()
+		require.NoError(t, err)
+
+		// Viper should handle the conversion from float to bool (0.0 -> false)
+		assert.Equal(t, false, param.Client_DisableHttpProxy.GetBool())
+	})
+
+	t.Run("job-ad-with-nested-classad", func(t *testing.T) {
+		ResetConfig()
+
+		tmpDir := t.TempDir()
+		jobAdFile := filepath.Join(tmpDir, "test.job.ad")
+
+		// Test with a nested ClassAd structure
+		jobAdContent := `PelicanCfg_Origin_Exports = [FederationPrefix = "/test"; StoragePrefix = "/storage"; Capabilities = {"Reads", "Writes"}]
+`
+		err := os.WriteFile(jobAdFile, []byte(jobAdContent), 0644)
+		require.NoError(t, err)
+
+		os.Setenv("_CONDOR_JOB_AD", jobAdFile)
+		t.Cleanup(func() {
+			os.Unsetenv("_CONDOR_JOB_AD")
+		})
+
+		bindClassAdConfig()
+		_, err = param.Refresh()
+		require.NoError(t, err)
+
+		// The nested structure should be accessible as a map
+		// Note: This test verifies the structure is parsed without errors
+		// Actual validation would depend on how Origin.Exports is used
```

Comment:
Yeah, do more here.  Make sure the resulting structure looks like what is expected.